wire-ios-sync-engine/Source/Synchronization/EventProcessor.swift

```diff
@@ -148,14 +151,20 @@ actor EventProcessor: UpdateEventProcessor {
 
             Logging.eventProcessing.info("Consuming: [\n\(decryptedUpdateEvents.map({ "\tevent: \(ZMUpdateEvent.eventTypeString(for: $0.type) ?? "Unknown")" }).joined(separator: "\n"))\n]")
 
-            await syncContext.perform {
-                for event in decryptedUpdateEvents {
+            for event in decryptedUpdateEvents {
+                await syncContext.perform {
                     for eventConsumer in self.eventConsumers {
-                        // TODO: [jacob] EventConsumer.processEvents should become async
                         eventConsumer.processEvents([event], liveEvents: true, prefetchResult: prefetchResult)
                     }
-                    self.eventProcessingTracker.registerEventProcessed()
                 }
+                // TODO: [F] @Jacob should this be done on syncContext to keep every thing in sync?
```

Comment:
We can't call an async function inside a `syncContext.perform {}`, but Isn't this already in sync? We are awaiting each call to `processEvents()`

Reply:
exactly, I thought this was not needed.
but I remember you mention something about respecting that other eventConsumers are supposed to run on syncContext

Reply:
Don't remember I think it's fine like this.